cosmetics-web/app/models/cpnp_export.rb

```
@@ -63,7 +63,7 @@ def shades
 
   def components
     current_version_component_lists_node.xpath(".//component").collect do |component_node|
-      Component.create(name: component_name(component_node),
+      Component.create!(name: component_name(component_node),
```

Is there a reason why we’d prefer to throw an exception if the object can’t be saved?

---

Yeah, I think we’d want to see an exception if this fails to save, so that we can see where any errors are in Sentry? Better than failing silently I think...?

---

Yeah, though the exception being thrown won't be handled cleanly and we'll still have to dig into which one of these components failed, and why, as this is looping 🤔 

What kind of exception are we expecting?

Maybe it's okay to have return nil here or some kind of sentinel, so that we have a rough idea of what actually went wrong.

---

Ok I've reverted for now, so it'll return a `nil`. This then causes the overall notification to fail to save when called from `ReadDataAnalyzer`, but it looks like there’s some existing error handling there.

The reason I added it was because we have an existing bug with parsing ZIP files containing components with PH ranges, but it was hard to debug because only a generic exception was being raised.

Let's revisit this at some point further down the line.